### PR TITLE
SDI-345: ✨ Initial stab at add restricted patient

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: integration_tests/integration/addPrisoner.cy.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubLogin', { roles: ['RESTRICTED_PATIENT_MIGRATION'] })
+    cy.task('stubAuthUser')
+    cy.task('stubUserRoles', [{ roleCode: 'RESTRICTED_PATIENT_MIGRATION' }])
+    cy.task('stubSearch', {
+      query: {
+        equalToJson: {
+          prisonerIdentifier: 'A1234AA',
+          prisonIds: ['OUT'],
```
Comment (petergphillips) [inline integration_tests/integration/addPrisoner.cy.js:23]: search queries on `OUT` prisoners

File: integration_tests/integration/homepage/homepage.cy.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ context('Homepage', () => {
 
       page.searchRestrictedPatient().should('exist')
       page.moveToHospital().should('not.exist')
-      page.addReleasedPrisoner().should('not.exist')
+      page.addRestrictedPatient().should('not.exist')
```
Comment (petergphillips) [inline integration_tests/integration/homepage/homepage.cy.js:28]: renamed from previous PR - add restricted patient makes more sense

File: integration_tests/pages/addPrisoner/addPrisonerSearch.page.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const page = require('../page')
+
+module.exports = {
+  verifyOnPage: () =>
+    page('Search for a prisoner to add', {
```
Comment (petergphillips) [inline integration_tests/pages/addPrisoner/addPrisonerSearch.page.js:5]: page title here contains `to add`

File: integration_tests/pages/movePrisoner/movePrisonerSearch.page.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ const page = require('../page')
 
 module.exports = {
   verifyOnPage: () =>
-    page('Search for a prisoner', {
+    page('Search for a prisoner to move', {
```
Comment (petergphillips) [inline integration_tests/pages/movePrisoner/movePrisonerSearch.page.js:5]: assert we're on a move page here

File: server/data/prisonerSearchResult.ts [hunk 5]
```diff
@@ -62,4 +62,10 @@ export default class PrisonerSearchResult {
 
   @Expose()
   alerts: AlertType[]
+
+  @Expose()
+  restrictedPatient: boolean
```
Comment (petergphillips) [inline server/data/prisonerSearchResult.ts:67]: grab extra data from prisoner search

File: server/routes/addPrisoner/index.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  const post = (path: string, handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
+
+  const prisonerSearch = new PrisonerSearchRoutes('/add-restricted-patient/select-prisoner')
+  const prisonerSelect = new PrisonerSelectRoutes(prisonerSearchService)
+
+  get('/search-for-prisoner', prisonerSearch.view)
+  post('/search-for-prisoner', prisonerSearch.submit)
+
+  get('/select-prisoner', prisonerSelect.view)
+  post('/select-prisoner', prisonerSelect.submit)
```
Comment (petergphillips) [inline server/routes/addPrisoner/index.ts:27]: at present only do the search and select flows - rest to follow

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**